sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/TransformWithStateExec.scala

```diff
@@ -161,7 +179,15 @@ case class TransformWithStateExec(
         procTimeIter.flatMap { case rowPair =>
           handleTimerRows(store, rowPair.key, batchTimestampMs.get)
         }
+      case EventTime =>
+        assert(batchTimestampMs.isDefined)
+        store.createColFamilyIfAbsent(TimerStateUtils.EVENT_TIMERS_STATE_NAME, true)
 
+        val eventTimeIter = store
+          .iterator(TimerStateUtils.EVENT_TIMERS_STATE_NAME)
+        eventTimeIter.flatMap { case rowPair =>
+          handleTimerRows(store, rowPair.key, batchTimestampMs.get)
```

Comment:
We can't pass `batchTimestampMs` here - thats only relevant for proc time. For event time, we need to check if the registered timestamp has expired compared to the eviction watermark timestamp.